openam-shared/src/main/java/com/sun/identity/shared/debug/Debug.java

```diff
@@ -259,38 +261,45 @@ static IDebugProvider getDebugProvider() {
      */
     private static synchronized void initialize() {
         if (!serviceInitialized) {
-            String providerName = SystemPropertiesManager.get(DebugConstants.CONFIG_DEBUG_PROVIDER);
+            final String providerName;
             IDebugProvider provider = null;
             boolean providerLoadFailed = false;
-            Exception exceptionCatched = null;
-            if (providerName != null && providerName.trim().length() > 0) {
+            Exception exceptionCaught = null;
+
+            providerName =
+                Optional.ofNullable(
+                    SystemPropertiesManager.get(DebugConstants.CONFIG_DEBUG_PROVIDER)
+                ).map(String::trim)
+                .orElse(null);
+
+            if (!Strings.isNullOrEmpty(providerName)) {
                 try {
-                    provider = (IDebugProvider) Class.forName(providerName).newInstance();
-                } catch (ClassNotFoundException cnex) {
+                    provider = (IDebugProvider)Class.forName(providerName).newInstance();
+                } catch (ClassNotFoundException
```

Comment:
Is there a point in catching specific exceptions? Doesn't any exception mean the provider failed to load? How would we deal with newer Java version throwing other exceptions?